src/lib/HasuraGraphiQL.tsx

```diff
@@ -0,0 +1,563 @@
+                      i,
+                      1,
+                      e.target.value
+                    );
+                    edited = edited2DArray(
+                      edited,
+                      i,
+                      3,
+                      hiddenHeaders.includes(e.target.value)
+                    );
+                    setHeadersInput(edited);
+                  }}
+                  className="hasura-graphiql-table-input"
+                  placeholder="Enter Key"
+                  type="text"
+                  data-testid={`row-key-${i}`}
+                  value={header[1]}
+                />
+              </td>
+              <td colSpan={1} className="hasura-graphiql-table-cell">
+                <input
+                  onBlur={updateHeaders}
+                  onChange={(e) =>
+                    setHeadersInput(
+                      edited2DArray(headersInput, i, 2, e.target.value)
+                    )
+                  }
+                  className="hasura-graphiql-table-input"
+                  placeholder="Enter Value"
+                  data-testid={`row-value-${i}`}
+                  type={header[3] ? "password" : "text"}
+                  value={header[2]}
+                />
+              </td>
+              <td className="hasura-graphiql-table-cell-cross">
+                {hiddenHeaders.includes(header[1]) && (
+                  <span
+                    style={{ marginRight: "1em" }}
+                    onClick={() => {
+                      let toggled = !header[3];
+                      setHeadersInput(
+                        edited2DArray(headersInput, i, 3, toggled)
+                      );
+                      setHeaders(edited2DArray(headersInput, i, 3, toggled));
+                    }}
+                  >
+                    <IconEye />
+                  </span>
+                )}
+                <i
+                  onClick={() => {
+                    let result = headersInput.slice();
+                    result.splice(i, 1);
+                    setHeadersInput(result);
+                    setHeaders(result);
+                    setLoading(true);
+                  }}
+                >
+                  <IconCross />
+                </i>
+              </td>
+            </tr>
+          ))}
+          <tr>
+            <td className="hasura-graphiql-table-cell-empty"></td>
+            <td className="hasura-graphiql-table-cell-key-entry">
+              <input
+                className="hasura-graphiql-table-cell-key-entry-input"
+                data-header-id="2"
+                placeholder="Enter Key"
+                data-element-name="key"
+                type="text"
+                data-test="header-key-2"
+                value=""
+                onChange={(e) => {
+                  setHeadersInput(
+                    headersInput.concat([[true, e.target.value, "", false]])
+                  );
+                  document
```

Comment:
Can we achieve this without dom manipulation?
Firstly, are we sure we need this? Secondly, even if this is needed, there might be more idiomatic ways to do this in React. Console does not do this IIUC.

Also `header-key-2` is hardcoded. We can definitely improve the name to something more readable. `header-key-placeholder` or something along those lines.

Reply:
Yes, with some redesign, it should be possible to do this with imperatively setting focus.